### PR TITLE
onReset empty

### PR DESCRIPTION
**Pull request by dahlbyk**
TL;DR: removes error elements in `onReset`.

It's not documented ([question](https://stackoverflow.com/questions/11360925/how-to-clear-jquery-validate-errors), [question](https://stackoverflow.com/questions/2086287/how-to-clear-jquery-validation-error-messages)), but one can generally reset unobtrusive state with `$(form).trigger('reset.unobtrusiveValidation')`:

https://github.com/aspnet/jquery-validation-unobtrusive/blob/74e6ccc7933ac224bdbef84b8dfff6d7ec19f286/src/jquery.validate.unobtrusive.js#L154-L159

This handily includes a call to [jQuery Validator `resetForm()`](https://jqueryvalidation.org/Validator.resetForm/):

https://github.com/aspnet/jquery-validation-unobtrusive/blob/74e6ccc7933ac224bdbef84b8dfff6d7ec19f286/src/jquery.validate.unobtrusive.js#L108

However, it seems like the logic to reset each individual error should account for the fact that it might contain an error message:

https://github.com/aspnet/jquery-validation-unobtrusive/blob/74e6ccc7933ac224bdbef84b8dfff6d7ec19f286/src/jquery.validate.unobtrusive.js#L116-L121

Similarly, resetting `.validation-summary-errors` will leave `<li>`s behind:

https://github.com/aspnet/jquery-validation-unobtrusive/blob/74e6ccc7933ac224bdbef84b8dfff6d7ec19f286/src/jquery.validate.unobtrusive.js#L113-L115

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
@ryanbrandenburg, @javiercn can you guys please review this change? Thanks!

**Comment by natemcmaster** [comment]
Updated branch protections since this repo is still using AppVeyor.

**Comment by ryanbrandenburg** [comment]
@dahlbyk could you provide a repo which demonstrates the issue this PR is trying to solve? That would help me understand better what we're fixing and what other consequences it might have.

**Comment by dahlbyk** [comment]
Use case is an add form (rendered in a modal, but not relevant) and submitted via AJAX. Model errors are parsed from a JSON response and passed through to `$form.validate().showErrors(errors)`; this plugin renders those errors as expected.

The problem is resetting the form on success to add another. Resetting jQuery Validator via `resetForm()` works as expected, but triggering `reset.unobtrusiveValidation` leaves behind the error `span`s inside a `.field-validation-valid` container, e.g.

```html
<span class="text-danger field-validation-valid" data-valmsg-for="Date" data-valmsg-replace="true">
 <span id="Date-error" class="">The value '123' is not valid for Date.</span>
</span>
```

I have worked around this with `.field-validation-valid { display: none; }`, but I'd prefer the DOM be truly reset.

**Comment by ryanbrandenburg** [comment]
I'm going to close this as we don't have a repro. @dahlbyk if you're still interested in getting this change in please open a new PR and include a link to a repo which demonstrates the problem you're trying to solve.
